.github/workflows/build.yml

```diff
@@ -4,16 +4,26 @@ on: [pull_request]
 
 jobs:
   build:
+    name: 🏭 Build all packages
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-      with:
-        ref: ${{ github.event.release.target_commitish }}
-    - name: Use Node.js 20
-      uses: actions/setup-node@v1
-      with:
-        node-version: 20
-        registry-url: https://registry.npmjs.org/
-    - run: npm ci
-    - run: npm run lint:check && npm run format:check
-    - run: npm run build
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Enable Corepack
+        run: corepack enable
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+
+      - name: Install
+        run: yarn install
+
+      - name: Build all packages
+        run: yarn build
```

Comment:
This should be `yarn nx run-many -t build` or the build script in the `package.json` should be, otherwise you aren't pulling any of the config defined in the `build` targets in each of the `project.json` files